### DISCUSSION TITLE
Allow custom messages in expect()

### DISCUSSION THREAD

**Discussion by clayreimann** [Quick/Nimble#900] [Ideas]
- [x] I have read [CONTRIBUTING](https://github.com/Quick/Nimble/blob/master/CONTRIBUTING.md) and have done my best to follow them.

### What did you do?

```swift
for (tag, title) in defaultRows {
    let row = entryForm.row(tag: tag)
    expect(row.exists) == true
    expect(row.staticTexts["title"].label) == title
}
```

### What did you expect to happen?

I would like to be able to provide custom message to `expect` so that I can see which loop iteration failed.

```swift
for (tag, title) in defaultRows {
    let row = entryForm.row(tag: tag)
    expect(row.exists, message: "\(tag) exists") == true
// or maybe more briefly
//    expect(row.exists, "\(tag) exists") == true
    expect(row.staticTexts["title"].label) == title
}
```

I'd be willing to submit a PR if this feature fits with the project

### Environment

List the software versions you're using:

 - Quick: Nope
 - Nimble: *7.1.3*
 - Xcode Version: *9.4.1 (9F2000)*
 - Swift Version: *Xcode Default* 
 - Carthage: *0.30.1*

**Comment by ikesyo** [comment]
The functionality is provided by `to(_:description:)`

https://github.com/Quick/Nimble/blob/f52c57e36830352f7e9ea95b4d3ce364572fea62/Tests/NimbleTests/UserDescriptionTest.swift#L12

**Comment by clayreimann** [comment]
Which is incompatible with the infix operators. Chai provides a [similar api](http://www.chaijs.com/api/bdd/#method_false) for their assertions that aren't function calls.

**Comment by clayreimann** [comment]
@ikesyo Any comment on this? Would such a PR be accepted?

**Comment by ikesyo** [comment]
It seems that this would be important with regard to #493.

**Comment by ikesyo** [comment]
I'm now working on this in relation to #369 and #742.
